crates/cli/src/commands/setup/windows.rs

```diff
@@ -0,0 +1,154 @@
+use std::ffi::OsString;
+use std::os::windows::ffi::OsStrExt;
+use std::path::PathBuf;
+
+use miette::{IntoDiagnostic, Result};
+
+use tracing::debug;
+use winreg::enums::{RegType, HKEY_CURRENT_USER, KEY_READ, KEY_WRITE};
+use winreg::{RegKey, RegValue};
+
+pub(crate) fn do_add_to_path(dirs: Vec<PathBuf>) -> Result<bool> {
+    let current_path = get_path_var()?;
+
+    let dirs: Vec<Vec<u16>> = dirs
+        .iter()
+        .map(|dir| OsString::from(dir).encode_wide().collect::<Vec<u16>>())
+        .filter(|dir| !path_contains(&current_path, &dir))
+        .collect();
+
+    let new_path = dirs
+        .iter()
+        .chain([&current_path])
+        .fold(vec![], |acc, path| path_join(&acc, &path));
+
+    if current_path == new_path {
+        debug!("System PATH already contains the new entries, leaving it untouched");
+        Ok(false)
+    } else {
+        debug!("Updating system PATH");
+        set_path_var(&new_path)?;
+        Ok(true)
+    }
+}
+
+fn get_path_var() -> Result<Vec<u16>> {
+    use std::io::{Error, ErrorKind};
+
+    RegKey::predef(HKEY_CURRENT_USER)
+        .open_subkey_with_flags("Environment", KEY_READ)
+        .and_then(|env| env.get_raw_value("PATH"))
+        .and_then(|value| {
+            winreg_ext::from_winreg_value(&value).ok_or(Error::new(
+                ErrorKind::InvalidData,
+                "The registry key HKEY_CURRENT_USER\\Environment\\PATH is not a string",
+            ))
+        })
+        .or_else(|err| {
+            if err.kind() == ErrorKind::NotFound {
+                Ok(vec![])
+            } else {
+                Err(err)
+            }
+        })
+        .into_diagnostic()
+}
+
+fn set_path_var(new_path: &[u16]) -> Result<()> {
+    RegKey::predef(HKEY_CURRENT_USER)
+        .open_subkey_with_flags("Environment", KEY_READ | KEY_WRITE)
+        .and_then(|environment| {
+            if new_path.is_empty() {
+                environment.delete_value("PATH")
```

Comment:
Does this delete the whole value? Seems risky no?

Reply:
Yeah it does, but if removed it would just be emptying the variable instead. Same effective result.

If you ever plan to have an uninstaller, then this would make sense to keep. But for now it's probably better I just make it error if the new_path is empty.

Reply:
Yeah it just seems odd. When would `new_path` be empty? Since we own that value and pass it in.

Reply:
True. Replaced that check with an assertion that it isn't empty.